### ISSUE TITLE
Missing region in config

### ISSUE THREAD

**Issue by davidascher** [lambci/cli#1]
I'm trying to set some env vars for a lambci deployment which seems to have been setup well, and I'm struggling to configure it with `lambci`:

```
mozilla-05039:api davida$ lambci info
CLI version: 0.2.2
Stack name: lambci
{ ConfigError: Missing region in config
    at Request.VALIDATE_REGION (/usr/local/lib/node_modules/lambci-cli/node_modules/aws-sdk/lib/event_listeners.js:81:45)
    at Request.callListeners (/usr/local/lib/node_modules/lambci-cli/node_modules/aws-sdk/lib/sequential_executor.js:105:20)
    at callNextListener (/usr/local/lib/node_modules/lambci-cli/node_modules/aws-sdk/lib/sequential_executor.js:95:12)
    at /usr/local/lib/node_modules/lambci-cli/node_modules/aws-sdk/lib/event_listeners.js:75:9
    at finish (/usr/local/lib/node_modules/lambci-cli/node_modules/aws-sdk/lib/config.js:307:7)
    at /usr/local/lib/node_modules/lambci-cli/node_modules/aws-sdk/lib/config.js:347:9
    at resolveNext (/usr/local/lib/node_modules/lambci-cli/node_modules/aws-sdk/lib/credentials/credential_provider_chain.js:84:9)
    at /usr/local/lib/node_modules/lambci-cli/node_modules/aws-sdk/lib/credentials/credential_provider_chain.js:97:11
    at /usr/local/lib/node_modules/lambci-cli/node_modules/aws-sdk/lib/credentials.js:123:23
    at /usr/local/lib/node_modules/lambci-cli/node_modules/aws-sdk/lib/credentials/ec2_metadata_credentials.js:66:7
  message: 'Missing region in config',
  code: 'ConfigError',
  time: 2016-07-06T23:03:13.067Z }
```

What config file is it looking for?  Is it the one that's in dynamo, and if so, how do I get it out of AWS into the project?

**Comment by mhart** [comment]
Ah yes, so this uses the AWS JavaScript SDK and needs to be configured as such. I should definitely mention that in the README.

Documentation for that is here: http://docs.aws.amazon.com/AWSJavaScriptSDK/guide/node-configuring.html

So I'd suggest either settings credentials in `~/.aws/credentials` or in environment variables

**Comment by davidascher** [comment]
Gotcha. I happen to be on a machine where i hadn't setup my .aws/credentials yet, but it makes a lot of sense.

**Comment by mhart** [comment]
I'll leave this open until I've had a chance to add it to the documentation 👍

**Comment by davidascher** [comment]
I await the docs.  I've configured ~/.aws/credentials (with access key secret access key) and ~/.aws/config (with region), and I'm still getting a validation error on region.  So something about my setup (which is good enough for the AWS cli tool) isn't good enough for the aws-sdk.  Investigating.

**Comment by mhart** [comment]
Hmmm, intriguing – I wonder if the SDK doesn't look in `~/.aws/config` for the region? I actually have mine set as an environment variable (`AWS_REGION`)

I might need to add a command line flag for the region specifically (and default to us-east-1)

**Comment by mhart** [comment]
Or I guess I could just document that ppl can set that env variable if they don't want us-east-1, and use it when they invoke, ie:

``` console
AWS_REGION=us-west-2 lambci config ...
```

**Comment by davidascher** [comment]
I think I figured it out.  https://github.com/lambci/cli/blob/master/github.js#L52 expects to find the environment variable, as the sdk doesn't specify the region.

Lots of ways around this, I think the only key thing is to guard against the mysterious error and put in a console output in case the environment var isn't set.

I _think_ the idiomatic thing is to use profiles defined in ~/.aws/config, and specify the regions in that file. See http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-config-files

**Comment by mhart** [comment]
Hmmm, I missed that – but that still wouldn't give you the error that you saw.

In any case, I wrote a library a while back to load both credentials and region from the config files for a separate project, so I think I'll use that – it covers nearly all credential and region config locations:

https://github.com/mhart/awscred

**Comment by mhart** [comment]
Just weird that the SDK doesn't load region:

https://github.com/aws/aws-sdk-js/issues/1039

**Comment by davidascher** [comment]
Agreed.  The usage examples all seem to expect the region to be configured via code, which is odd.  I expect it'll get fixed.

**Comment by mhart** [comment]
Thanks for reporting this @davidascher – should be fixed with the latest version (v0.2.3) – and I've added some documentation to the README about configuring AWS params
